libyelp/yelp-settings.c

```diff
@@ -599,11 +599,18 @@ yelp_settings_get_default (void)
     static GMutex mutex;
     static YelpSettings *settings = NULL;
     g_mutex_lock (&mutex);
-    if (settings == NULL)
-	settings = g_object_new (YELP_TYPE_SETTINGS,
-				 "gtk-settings", gtk_settings_get_default (),
-				 "gtk-icon-theme", gtk_icon_theme_get_default (),
-				 NULL);
+    if (settings == NULL) {
```

Comment:
The `settings == NULL` check was there before.

Checked the rest, we still need it despite the gtk_init.